quinn-proto/src/connection/mtud.rs

```diff
@@ -336,136 +335,178 @@ impl SearchState {
     }
 }
 
+/// Judges whether packet loss might indicate a drop in MTU
+///
+/// Our MTU black hole detection scheme is a heuristic based on the order in which packets were sent
+/// (the packet number order), their sizes, and which are deemed lost. We track the smallest packet
```

Comment:
Elaborated some.

Reply:
So here's something that I was thinking about yesterday: what if there is an MTU reduction and we only receive larger packets from the peer? The loss burst never "ends", so we never detect a black hole (because that is contingent on finding 3 suspicious loss bursts). Is that an issue? If not, why not?

I'd structure this more like this:

```rust
/// Our MTU black hole detection scheme is a heuristic based on the order in which packets were sent
/// (the packet number order), their sizes, and which are deemed lost.
///
/// We try to detect contiguous groups of lost packets (a "loss burst"), which we aggregate
/// because a group of packets all lost together were probably lost for the same reason.
///
/// A loss burst is deemed "suspicious" if it contains no packets that are (a) smaller than the
/// minimum MTU or (b) smaller than an acknowledged more recent packet, because such a burst
/// could be fully explained by a reduction in MTU.
///
/// When the number of suspicious loss bursts exceeds [`BLACK_HOLE_THRESHOLD`] [because ...?],
/// detection of a black hole is signaled.
```

This tries to more clearly introduce the concepts used/introduced in this type.

Reply:
> what if there is an MTU reduction and we only receive larger packets from the peer?

You mean, what if we're only sending larger packets? If we're not getting any ACKs, we'll eventually send a tail loss probe. Right now it looks like those are necessarily below the minimum MTU by coincidence (we never bothered to hook up the capability to include application data in loss probes), but we should probably enforce that. If tail loss probes are small, then they'll be ACKed and break up an uninterrupted loss stream.

Reply:
But can we count on the ACK for the tail loss probe to be small? Is the peer required to avoid filling up that packet with stream data or whatever?

Reply:
We have no control over the size of packets constructed by the peer; this logic is entirely concerned with the size of the packets we sent, and whether or not they're ACKed. The peer should be running its own black hole detection and, if it *is* sending us lots of stream data, notice the losses and generate its own tail loss and/or MTU probes accordingly.

Reply:
Oh, right, I guess that's my disconnect: this is ultimately about packets we sent, not packets we receive, so we do have control over whether/when we send shorter packets. I guess for received packets we don't know the packet sizes for lost packets anyway so it's hard to use that info.

Reply:
Also, if the peer is losing lots of packets it'll quickly become congestion-bound, but ACK-only packets aren't congestion controlled, so they should generally come through on their own.

Reply:
Reworded the comment and introduced a separate commit to limit tail loss probe size as discussed.